TSheetsApi/UserAuthentication.cs

```diff
@@ -123,10 +123,11 @@ public OAuthToken GetToken()
             }
             else if (_token.NeedsRefresh())
             {
+                var refreshToken = _token.refresh_token;
                 _token = null;
                 try
                 {
-                    _token = RestClient.RefreshToken(_token.refresh_token, _connectionInfo);
+                    _token = RestClient.RefreshToken(refreshToken.refresh_token, _connectionInfo);
```

Comment:
I think we need to make one small change. We saved off just the refreshToken value of the OAuthToken object @ line 126, so we just need to provide that value in the refresh call here:
```suggestion
                    _token = RestClient.RefreshToken(refreshToken, _connectionInfo);
```

Reply:
Gah, I knew I'd make a stupid mistake after attempting the revision/PR process twice, incorrectly, before finally remembering that I should fork, not clone. That, or I crossed wires in what I would have done, versus what you suggested - I'm pretty sure I stored off the entire token when I was testing over the weekend, because I verified the request looked complete, including the refresh token.  I'll make this change, sorry for the simple mistake.